### PR TITLE
CAS-284 import device traversal code

### PR DESCRIPTION
**Pull request by gila**
This code is based on older device inventory code. We dont need the
inventory bits, but this code allows us to lookup a device based in the
storage URI's such that we can find devices based on the UUIDs.

### CODE REVIEW

File: devinfo/src/block_device.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+        let uuid = Uuid::parse_str(nq[1]).map_err(|e| NqnInvalid {
+            value: format!("the UUID is invalid {}", e.to_string()),
+        })?;
+
+        match value.scheme() {
+            "iscsi" => Ok(BlkDev::Scsi(uuid)),
+            "nvmf" => Ok(BlkDev::Nvmf(uuid)),
+            _ => Err(DevInfoError::NotSupported {
+                value: "".to_string(),
```
Comment (paulyoong) [inline devinfo/src/block_device.rs:68]: Should we return the scheme that isn't supported here?

File: devinfo/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+mod block_device;
+use snafu::Snafu;
+
+#[derive(Debug, Snafu)]
+pub enum DevInfoError {
+    #[snafu(display("Device {} not found", path))]
+    NotFound { path: String },
+    #[snafu(display("Failed to parse value {}", value))]
+    ParseError { value: String },
+    #[snafu(display(" Name qualifier invalid {} ", value))]
```
Comment (paulyoong) [inline devinfo/src/lib.rs:13]: Erroneous space before "Name"

Comment (gila) [reply devinfo/src/lib.rs:13]: :+1:

File: devinfo/src/block_device.rs (hunk 1 above)
Comment (gila) [reply devinfo/src/block_device.rs:68]: Thanks! fixed :+1:

**Pull request merged**